go/feast/featurestore.go

```diff
@@ -0,0 +1,907 @@
+package feast
+
+import (
+	"errors"
+	"fmt"
+	"github.com/feast-dev/feast/go/protos/feast/serving"
+	"github.com/feast-dev/feast/go/protos/feast/types"
+	durationpb "google.golang.org/protobuf/types/known/durationpb"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+	"sort"
+	"strings"
+)
+
+type FeatureStore struct {
+	config      *RepoConfig
+	registry    *Registry
+	onlineStore OnlineStore
+}
+
+type entityKeyRow struct {
+	entityKey  *types.EntityKey
+	rowIndices []int
+}
+
+// NewFeatureStore constructs a feature store fat client using the
+// repo config (contents of feature_store.yaml converted to JSON map).
+func NewFeatureStore(config *RepoConfig) (*FeatureStore, error) {
+	onlineStore, err := getOnlineStore(config)
```

Comment:
Would it make sense to use `NewOnlineStore` here in order to stay consistent with the rest of the code base?